docs/source/aggregation.rst

```diff
@@ -0,0 +1,182 @@
+.. _aggregation:
+
+===========
+Aggregation
+===========
+
+This chapter provides information about the passing aggregates of jobs to operation functions.
+
+
+.. _aggregator_definition:
+
+aggregator
+==========
+
+An :class:`~flow.aggregator` is used as a decorator for operation functions which accept a variable number of positional arguments, ``*jobs``.
+The argument ``*jobs`` is unpacked into an *aggregate*, defined as an ordered tuple of jobs.
+See also the Python documentation about :ref:`argument unpacking <python:tut-unpacking-arguments>`.
+
+.. code-block:: python
+
+    # project.py
+    from flow import FlowProject, aggregator
+
+    class Project(FlowProject):
+        pass
+
+    @aggregator()
+    @Project.operation
+    def op1(*jobs):
+        print("Number of jobs in aggregate:", len(jobs))
+
+    @Project.operation
+    def op2(job):
+        pass
+
+    if __name__ == '__main__':
+        Project().main()
+
+If :class:`~flow.aggregator` is used with the default arguments, it will create a single aggregate containing all the jobs present in the project.
+In the example above, ``op1`` is an *aggregate operation* where all the jobs present in the project are passed as a variable number of positional arguments (via ``*jobs``), while ``op2`` is a normal operation where only a single job is passed as an argument.
```

Comment:
```suggestion
In the example above, ``op1`` is an *aggregate operation* where all the jobs present in the project are passed as a variable number of positional arguments (via ``*jobs``), while ``op2`` is an operation where only a single job is passed as an argument.

.. tip:

    The concept of aggregation might be easier to understand if one realizes that "normal" operation functions are equivalent to aggregate-operations-functions with an aggregate group size of one.
```
I don't like the distinction between "aggregate" and "normal" operation functions as it implies that aggregate operations are somehow not normal. However, that's not the case. Every operation function is an aggregate operation function, just that "normal operations" are in the limit of aggregate size one. I already explained previously in the context of documenting "groups": It is important to not introduce special cases, but instead try to relate the concepts to each other in a logical way.

- For a user it is easier to make the transfer from "normal" operations to operation groups, when they realize that each operation constitutes an operation group with a single entry.
- Similarly, I believe that a user will have an easier time to understand the aggregate-concept, when they realize that "normal operation functions" are aggregate operation functions of size 1.

In summary: in my view all operation functions are in principle the same, just by default they form a group of size 1 and an aggregate of size 1.

Reply:
I agree with this approach. I will leave this comment thread open until I have a chance to review again and see if there are other instances of this distinction. If this is the only place, we can just commit the suggested change.

Reply:
@csadorf Do you want to make an section that explicitly describes your point about all operations acting with aggregation in the limit of one job? I think that could strengthen the docs.

Reply:
> Do you want to make an section that explicitly describes your point about all operations acting with aggregation in the limit of one job?

I re-read the original comment. The suggestion posted by @csadorf includes exactly the language that I was hoping to add. Somehow I missed that the first time. 😅